### PR TITLE
Fixes #23898 - Allow setting per_page of hosts table

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

File: app/controllers/job_invocations_controller.rb [hunk 1]
```diff
@@ -57,7 +57,8 @@ def show
       hosts_base = hosts_base.joins(:template_invocations)
                              .where(:template_invocations => { :job_invocation_id => @job_invocation.id})
     end
-    @hosts = hosts_base.search_for(params[:search], :order => params[:order] || 'name ASC').paginate(:page => params[:page])
+    @hosts = hosts_base.search_for(params[:search], :order => params[:order] || 'name ASC')
```
Comment (ohadlevy) [inline app/controllers/job_invocations_controller.rb]: any reason why not use https://github.com/theforeman/foreman/blob/2b0401fc35c5aa0d4e059aba434ade4aafeae2a9/app/controllers/application_controller.rb#L224 ?

Comment (adamruzicka) [reply app/controllers/job_invocations_controller.rb]: Mainly that I didn't know we have something like that

File: app/controllers/job_invocations_controller.rb [hunk 2]
```diff
@@ -51,13 +51,13 @@ def create
...
+    @resource_base = @job_invocation.targeting.hosts.authorized(:view_hosts, Host)
     # There's no need to do the joining if we're not filtering
     unless params[:search].nil?
-      hosts_base = hosts_base.joins(:template_invocations)
-                             .where(:template_invocations => { :job_invocation_id => @job_invocation.id})
+      @resource_base = @resource_base.joins(:template_invocations)
+                                     .where(:template_invocations => { :job_invocation_id => @job_invocation.id})
     end
-    @hosts = hosts_base.search_for(params[:search], :order => params[:order] || 'name ASC').paginate(:page => params[:page])
+    @hosts = resource_base_search_and_page
```
Comment (ohadlevy) [inline app/controllers/job_invocations_controller.rb]: might consider changing index action too?

**Comment by iNecas** [comment]
[test foreman]

**Comment by iNecas** [comment]
[test foreman_remote_execution]

**Comment by adamruzicka** [comment]
[test foreman_remote_execution]

**Comment by iNecas** [comment]
@adamruzicka mind rebasing so that we fix the rubocop issues?

**Comment by adamruzicka** [comment]
Not at all, rebased

**Comment by adamruzicka** [comment]
Mr. J approves

**Comment by iNecas** [comment]
Thanks @adamruzicka

**Pull request merged**